### PR TITLE
Referrer Trimming test page

### PR DESCRIPTION
**Pull request by kdzwinel**
Live: https://privacy-test-pages.glitch.me/privacy-protections/referrer-trimming/
Tech Design: https://app.asana.com/0/481882893211075/1198919147577400/f

### CODE REVIEW

File: privacy-protections/referrer-trimming/main.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+const results = {
+    page: 'referrer-trimming',
+    date: null,
+    results: []
+};
+
+// list of localStorage entries with partial test results that we need to clear at the end of all testing
+const lsEntriesToClear = [];
+
+function generateNavigationTest(url) {
```
Comment (kdzwinel) [inline privacy-protections/referrer-trimming/main.js:18]: Some tests are very similar with only URL being different (1p, 3p, tracker), so I extracted their generation into functions.

File: privacy-protections/referrer-trimming/main.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        return result;
+    } else {// test haven't run yet
+        window.location.href = url;
+
+        // let test runner know that it should not run other tests
+        return 'stop';
+    }
+}
+
+function generateFrameTest(url) {
```
Comment (kdzwinel) [inline privacy-protections/referrer-trimming/main.js:50]: ditto

**Review by kdzwinel — COMMENTED** [review]
@Charlie-belmer  LMK what you think!

**Review by Charlie-belmer — APPROVED** [review]
Everything looks great!

**Pull request merged**